### PR TITLE
History setup

### PR DESCRIPTION
**Pull request by youthh**
added history & highest score & validation

### CODE REVIEW

File: src/Components/Alerts/Alert.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import "./AlertStyle.css";
+
+type AlertProps = {
+  isShowModalAlertMessage: boolean;
+};
+
+const Alerts = ({ isShowModalAlertMessage }: AlertProps) => {
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
```
Comment (krymchuk) [inline src/Components/Alerts/Alert.tsx:14]: what the reason for useEffect without any dependencies?
try to optimize in using isShowModalAlertMessage as dependency

File: src/Components/Alerts/AlertStyle.css [hunk 2]
```diff
@@ -0,0 +1,12 @@
+.AlertBox {
+  position: fixed;
+  z-index: 10000;
+  right: 20px;
+  top: 30px;
+}
+
+.boxAlert {
```
Comment (krymchuk) [inline src/Components/Alerts/AlertStyle.css]: what difference btwn AlertBox and boxAlert names?

File: src/Components/Alerts/AlertStyle.css [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  z-index: 10000;
+}
+
+/*# sourceMappingURL=AlertStyle.css.map */
```
Comment (krymchuk) [inline src/Components/Alerts/AlertStyle.css]: remove this comment

File: src/Components/Alerts/AlertStyle.css.map [hunk 4]
```diff
@@ -0,0 +1 @@
+{"version":3,"sourceRoot":"","sources":["AlertStyle.scss"],"names":[],"mappings":"AAAA;EACE;EACA;EACA;EACA;;;AAGF;EACE","file":"AlertStyle.css"}
```
Comment (krymchuk) [inline src/Components/Alerts/AlertStyle.css.map]: not sure you need it here

File: src/Components/Alerts/AlertStyle.scss [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  top: 30px;
+}
+
+.boxAlert{
+  z-index: 10000;
+
+}
+
+
+.activeAlert{
```
Comment (krymchuk) [inline src/Components/Alerts/AlertStyle.scss]: remove empty class

File: src/Components/Alerts/AlertStyle.scss [hunk 6]
```diff
@@ -0,0 +1,16 @@
+.AlertBox{
+  position: fixed;
+  z-index: 10000;
+  right: 20px;
+  top: 30px;
+}
+
+.boxAlert{
```
Comment (krymchuk) [inline src/Components/Alerts/AlertStyle.scss]: DRY. this styles is duplicated.

File: src/Components/Header/header.css.map [hunk 7]
```diff
@@ -1 +1 @@
-{"version":3,"sourceRoot":"","sources":["header.scss"],"names":[],"mappings":"AAAA;EAEE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAIF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAIF;EACE;EACA;;;AAKF;EACE;EACA;;AACA;EACE;;;AAIJ;EACE;EACA;EACA","file":"header.css"}
\ No newline at end of file
+{"version":3,"sourceRoot":"","sources":["header.scss"],"names":[],"mappings":"AAAA;EAEE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAIF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAIF;EACE;EACA;;;AAKF;EACE;EACA;;AACA;EACE;;;AAIJ;EACE;EACA;EACA;;;AAGF;EACE;EACA;;;AAGF;EACE;EACA;EACA","file":"header.css"}
```
Comment (krymchuk) [inline src/Components/Header/header.css.map]: remove this map file

File: src/Components/Header/header.css [hunk 8]
```diff
@@ -56,4 +56,15 @@
...
+  height: 58px;
+}
+
+.left_header {
+  display: flex;
+  align-items: center;
+  min-height: 82px;
+}
+
 /*# sourceMappingURL=header.css.map */
```
Comment (krymchuk) [inline src/Components/Header/header.css]:
```suggestion
```

File: src/Components/Header/header.scss [hunk 9]
```diff
@@ -60,3 +60,14 @@
   align-items: center;
   min-height: 82px;
 }
+
+.header__img_avatar{
+  width: 60px;
+  height: 58px;
+}
+
+.left_header{
```
Comment (krymchuk) [inline src/Components/Header/header.scss:69]: duplicated styles. Use only scss files for styles declaration

File: src/Components/LeaderBoard/LeaderBoard.scss [hunk 10]
```diff
@@ -1,16 +1,25 @@
...
+  overflow: hidden;
   display: flex;
-  justify-content: center;
+  justify-content: flex-start;
   align-items: center;
   flex-direction: column;
 }
 
-.leaderBoard_container--top{
+.boxSpinner{
```
Comment (krymchuk) [inline src/Components/LeaderBoard/LeaderBoard.scss]:
```suggestion
.boxSpinner {
```

File: src/Components/LeaderBoard/LeaderBoard.scss [hunk 11]
```diff
@@ -1,16 +1,25 @@
...
+  transition: .3s;
+}
+
+
+.boxSpinner--active{
```
Comment (krymchuk) [inline src/Components/LeaderBoard/LeaderBoard.scss]:
```suggestion
.boxSpinner--active {
```

File: src/Components/LeaderBoard/LeaderBoard.tsx [hunk 12]
```diff
@@ -2,29 +2,52 @@ import React, { useEffect } from "react";
...
 
 const LeaderBoard = () => {
   const dispatch = useAppDispatch();
-  const { leaderBoard, isLoading } = useAppSelector(leaderSelector);
+  const { leaderBoard, page, isAnotherDayLoading } =
+    useAppSelector(leaderSelector);
+
+  const changePage = (a: number) => {
+    dispatch(setLeadersLoading(true));
+    setTimeout(() => {
```
Comment (krymchuk) [inline src/Components/LeaderBoard/LeaderBoard.tsx]: why you need set timeout for loading stop?

File: src/Components/Logic/logicLeaderSlice.ts [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+const setLeaderField = (leader: Leader, index: number) => {
+  return Object.assign(leader, {
+    img: avatars[index],
+    color: "#F99746",
+    changesCount: 0,
+    score: Object.hasOwn(leader, "score") ? leader.score : 0,
+  });
+};
+const randomColor = [
```
Comment (krymchuk) [inline src/Components/Logic/logicLeaderSlice.ts]: move this array to another file and import here

File: src/Components/Modal/Modal.tsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+const BasicModal = ({ children }: ModalProps) => {
+  const { isShowModal, isShowModalAlertMessage } = useSelector(modalSelector);
+  return (
+    <div>
+      <Modal
+        open={isShowModal}
+        aria-labelledby="modal-modal-title"
+        aria-describedby="modal-modal-description"
+      >
+        <Box sx={style}>{children.map((child) => child)}</Box>
```
Comment (krymchuk) [inline src/Components/Modal/Modal.tsx]: do not map children. Use fragments to wrap content in parent component like `<> .... </>`

File: src/Components/Modal/ModalEditScore.tsx [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+          color: "#030327",
+          boxShadow: "0px 2px 8px rgba(249, 151, 70, 0.04)",
+        }}
+        variant="contained"
+      >
+        Save
+      </Button>
+    </form>
+  ) : (
+    <></>
```
Comment (krymchuk) [inline src/Components/Modal/ModalEditScore.tsx]: just return null

File: src/app.scss [hunk 16]
```diff
@@ -15,7 +15,9 @@ body{
 }
 
 .inner__app{
-  max-width: 810px;
+  max-width: 840px;
   width: 100%;
   margin: 0 auto;
 }
+
+
```
Comment (krymchuk) [inline src/app.scss]:
```suggestion
```

File: src/api/leadersService.ts [hunk 17]
```diff
@@ -0,0 +1,39 @@
...
+    })
+    .catch(() => {
+      return addNewLeader(name);
+    });
+};
+
+export const getAnotherDayLeaders = (): Promise<Leader[]> => {
+  return instance
+    .get("starting-state")
+    .then((data) => {
```
Comment (krymchuk) [inline src/api/leadersService.ts]: .then((response) => resposne.data)

**Review by krymchuk — CHANGES_REQUESTED** [review]

**Comment by youthh** [comment]
added history & made review made refactoring &  made modal

**Pull request merged**